packages/sanity/src/core/store/events/getDocumentChanges.ts

```diff
@@ -0,0 +1,261 @@
+import {diffInput, wrap} from '@sanity/diff'
+import {type SanityDocument, type TransactionLogEventWithEffects} from '@sanity/types'
+import {applyPatch, incremental} from 'mendoza'
+import {
+  combineLatest,
+  from,
+  map,
+  type Observable,
+  of,
+  shareReplay,
+  startWith,
+  switchMap,
+  tap,
+} from 'rxjs'
+import {type SanityClient} from 'sanity'
+
+import {type Annotation, type ObjectDiff} from '../../field'
+import {wrapValue} from '../_legacy/history/history/diffValue'
+import {getDocumentTransactions} from './getDocumentTransactions'
+import {
+  type DocumentGroupEvent,
+  type EventsStoreRevision,
+  isCreateDocumentVersionEvent,
+} from './types'
+import {type EventsObservableValue} from './useEventsStore'
+
+const buildDocumentForDiffInput = (document?: Partial<SanityDocument> | null) => {
+  if (!document) return {}
+  // Remove internal fields and undefined values
+  const {_id, _rev, _createdAt, _updatedAt, _type, _version, ...rest} = JSON.parse(
+    JSON.stringify(document),
+  )
+
+  return rest
+}
+
+type EventMeta = {
+  transactionIndex: number
+  event?: DocumentGroupEvent
+} | null
+
+function omitRev(document: SanityDocument): Omit<SanityDocument, '_rev'> {
+  const {_rev, ...doc} = document
+  return doc
+}
+
+function annotationForTransactionIndex(
+  transactions: TransactionLogEventWithEffects[],
+  idx: number,
+  event?: DocumentGroupEvent,
+) {
+  const tx = transactions[idx]
+  if (!tx) return null
+
+  return {
+    timestamp: tx.timestamp,
+    author: tx.author,
+    event: event,
+  }
+}
+
+function extractAnnotationForFromInput(
+  transactions: TransactionLogEventWithEffects[],
+  meta: EventMeta,
+): Annotation {
+  if (meta) {
+    // The next transaction is where it disappeared:
```

Comment:
Forgive me, might be missing some context. What disappears? 🤔

Reply:
removing this, it doesn't make sense. 
This is how you go from **rev2** to **rev1** , so you want to use the transaction from **rev1**, not the one from **rev2**. 

I'm not even so sure on how this works in the internals. This is how it was implemented in the timeline diff generator and it works also here because we are getting the same data.